notebooks/Transform_data.md

```diff
@@ -59,28 +60,29 @@ print(f"Filtered columns: {df_filtered.memory_usage().sum()//10**6} mb")
 
 ```python
 @timer
-def transform_dataset(path_to_parquet, codes=None):
+def transform_dataset(path_to_parquet, codes=None, teams_df=None):
     if not codes:
         codes = pd.read_parquet(path_to_parquet, columns=["code"], partitioning=["team_code", "code"])
-    for i, code in enumerate(tqdm(df_selected["code"].unique()[:1])):
+    for i, code in enumerate(tqdm(df_selected["code"].unique())):
         df = pd.read_parquet(path_to_parquet, filters=[('code', '=', 98747)], partitioning=["team_code","code"])
-        df_g = df.groupby(by=["code", "gameweek"]).last()
-
+        df_g = df.groupby(by=["code", "gameweek"], as_index=False).last()
         # Before transformation
-        print(df_g["minutes"])
-        print(type(df_g["minutes"]))
-
+        df_g.loc[:, "minutes"] = calculate_diff(df_g["minutes"])
+        df_g = join_elements_and_team(df_g, teams_df)
         # After transformation
-        print(calculate_diff(df_g["minutes"]))
 ```
 
 ```python
-transform_dataset("../data/interim/2020_elements_parquet/")
+@timer
+def join_elements_and_team(elements_df, teams_df):
+    elements_df = elements_df.groupby(by=["code", "gameweek"], as_index=False).last()
```

Comment:
Jeg har ikke testet det før, men hva skjer hvis du kaller `groupby` med de samme kolonnene to ganger på rad mot samme dataframe? Det er vel det som skjer her (L68 og L78)?

Reply:
@mrminy det skal i utgangspunktet ikke skje noen ting. Koden her er forøvrig skrevet mens jeg prøvde å finne en workaround til følgende snåle feil. https://github.com/pandas-dev/pandas/issues/44734 

Ser selv i etterkant at det har blitt unødvendig mye rot.